### PR TITLE
Use PHP constants for doctrine config

### LINKED ISSUE #33524: Error installing version 8.1 SSL connection error: SSL_CTX_set_default_verify_pat
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

During the installation of version 8.1 on a subdomain, created only for that i get the following error.

- clean sub-domain and new sql database created - tested passed on connection on the step before.

image

```
1: 0
2: 1
[WARNING] Some commands could not be registered:
3: 2
4: 3
In ConnectionFactory.php line 142:
5: 4
6: 5
An exception occurred while establishing a connection to figure out your pl
7: 6
atform version.
8: 7
You can circumvent this by setting a 'server_version' configuration value
9: 8
10: 9
For further information have a look at:
11: 10
https://github.com/doctrine/DoctrineBundle/issues/673
12: 11
13: 12
14: 13
In AbstractMySQLDriver.php line 128:
15: 14
16: 15
An exception occurred in driver: SQLSTATE[HY000] [2026] SSL connection erro
17: 16
r: SSL_CTX_set_default_verify_paths failed
18: 17
19: 18
20: 19
In Exception.php line 18:
21: 20
22: 21
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
23: 22
hs failed
24: 23
25: 24
26: 25
In PDOConnection.php line 40:
27: 26
28: 27
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
29: 28
hs failed
30: 29
31: 30
32: 31
In Configuration.php line 331:
33: 32
34: 33
Argument 1 passed to PrestaShop\PrestaShop\Adapter\Configuration::restrictU
35: 34
pdatesTo() must be an instance of Shop, null given, called in /home/b001000
36: 35
0/public_html/subdomain.xxxx.com/src/Core/Addon/Theme/ThemeMa
37: 36
nagerBuilder.php on line 61
38: 37
39: 38
40: 39
In ConnectionFactory.php line 142:
41: 40
42: 41
An exception occurred while establishing a connection to figure out your pl
43: 42
atform version.
44: 43
You can circumvent this by setting a 'server_version' configuration value
45: 44
46: 45
For further information have a look at:
47: 46
https://github.com/doctrine/DoctrineBundle/issues/673
48: 47
49: 48
50: 49
In AbstractMySQLDriver.php line 128:
51: 50
52: 51
An exception occurred in driver: SQLSTATE[HY000] [2026] SSL connection erro
53: 52
r: SSL_CTX_set_default_verify_paths failed
54: 53
55: 54
56: 55
In Exception.php line 18:
57: 56
58: 57
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
59: 58
hs failed
60: 59
61: 60
62: 61
In PDOConnection.php line 40:
63: 62
64: 63
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
65: 64
hs failed
66: 65
67: 66
68: 67
69: 68
[WARNING] Some commands could not be registered:
70: 69
71: 70
In ConnectionFactory.php line 142:
72: 71
73: 72
An exception occurred while establishing a connection to figure out your pl
74: 73
atform version.
75: 74
You can circumvent this by setting a 'server_version' configuration value
76: 75
77: 76
For further information have a look at:
78: 77
https://github.com/doctrine/DoctrineBundle/issues/673
79: 78
80: 79
81: 80
In AbstractMySQLDriver.php line 128:
82: 81
83: 82
An exception occurred in driver: SQLSTATE[HY000] [2026] SSL connection erro
84: 83
r: SSL_CTX_set_default_verify_paths failed
85: 84
86: 85
87: 86
In Exception.php line 18:
88: 87
89: 88
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
90: 89
hs failed
91: 90
92: 91
93: 92
In PDOConnection.php line 40:
94: 93
95: 94
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
96: 95
hs failed
97: 96
98: 97
99: 98
In Configuration.php line 331:
100: 99
101: 100
Argument 1 passed to PrestaShop\PrestaShop\Adapter\Configuration::restrictU
102: 101
pdatesTo() must be an instance of Shop, null given, called in /home/b001000
103: 102
0/public_html/subdomain.xxxxxx.com/src/Core/Addon/Theme/ThemeMa
104: 103
nagerBuilder.php on line 61
105: 104
106: 105
107: 106
In ConnectionFactory.php line 142:
108: 107
109: 108
An exception occurred while establishing a connection to figure out your pl
110: 109
atform version.
111: 110
You can circumvent this by setting a 'server_version' configuration value
112: 111
113: 112
For further information have a look at:
114: 113
https://github.com/doctrine/DoctrineBundle/issues/673
115: 114
116: 115
117: 116
In AbstractMySQLDriver.php line 128:
118: 117
119: 118
An exception occurred in driver: SQLSTATE[HY000] [2026] SSL connection erro
120: 119
r: SSL_CTX_set_default_verify_paths failed
121: 120
122: 121
123: 122
In Exception.php line 18:
124: 123
125: 124
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
126: 125
hs failed
127: 126
128: 127
129: 128
In PDOConnection.php line 40:
130: 129
131: 130
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
132: 131
hs failed
133: 132
134: 133
135: 134
19:00:07 CRITICAL [console] Error thrown while running command "prestashop:schema:update-without-foreign --env=prod". Message: "An exception occurred while establishing a connection to figure out your platform version.
136: 135
You can circumvent this by setting a 'server_version' configuration value
137: 136
138: 137
For further information have a look at:
139: 138
https://github.com/doctrine/DoctrineBundle/issues/673" ["exception" => Doctrine\DBAL\Exception { …},"command" => "prestashop:schema:update-without-foreign --env=prod","message" => """ An exception occurred while establishing a connection to figure out your platform version.\n You can circumvent this by setting a 'server_version' configuration value\n \n For further information have a look at:\n https://github.com/doctrine/DoctrineBundle/issues/673 """]
140: 139
141: 140
In ConnectionFactory.php line 142:
142: 141
143: 142
An exception occurred while establishing a connection to figure out your pl
144: 143
atform version.
145: 144
You can circumvent this by setting a 'server_version' configuration value
146: 145
147: 146
For further information have a look at:
148: 147
https://github.com/doctrine/DoctrineBundle/issues/673
149: 148
150: 149
151: 150
In AbstractMySQLDriver.php line 128:
152: 151
153: 152
An exception occurred in driver: SQLSTATE[HY000] [2026] SSL connection erro
154: 153
r: SSL_CTX_set_default_verify_paths failed
155: 154
156: 155
157: 156
In Exception.php line 18:
158: 157
159: 158
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
160: 159
hs failed
161: 160
162: 161
163: 162
In PDOConnection.php line 40:
164: 163
165: 164
SQLSTATE[HY000] [2026] SSL connection error: SSL_CTX_set_default_verify_pat
166: 165
hs failed
167: 166
168: 167
169: 168
prestashop:schema:update-without-foreign [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [-e|--env ENV] [--no-debug] [--id_shop [ID_SHOP]] [--id_shop_group [ID_SHOP_GROUP]] [--]
170: 169
171: 170
```

**Server provider specs:**

- Apache Version | 2.4.57
- MySQL Version | 8.0.34
- Architecture | x86_64
- Operating System | linux
- Shared IP Address | xxxxxxxxxxxxxxxxxxx
- Path to Sendmail | /usr/sbin/sendmail
- Path to Perl | /usr/bin/perl
- Perl Version | 5.26.3
- Kernel Version | 4.18.0-348.7.1.lve.el8.x86_64

### Expected behavior

Normal Installation....

### Steps to reproduce

1. Put the zip and 2 files on  clean server folder
2. Start installation process.
3. Fill the forms with no issues.
4. Didn´t choose to install demo products
5. Install all suggested modules.

### PrestaShop version(s) where the bug happened

8.1

### PHP version(s) where the bug happened

7.4

### If your bug is related to a module, specify its name and its version

_No response_

### Your company or customer's name goes here (if applicable).

_No response_

### PR DESCRIPTION
**Pull request by matks**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Use PHP constants for doctrine config as informations collected in https://github.com/PrestaShop/PrestaShop/issues/33524 indicate the value of `PDO::MYSQL_ATTR_MULTI_STATEMENTS` change on different environments
| Type?             | bug fix
| Category?         | CO
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | Please see below
| UI Tests          | I will add the UI tests when I have fixed the CI
| Fixed issue or discussion?     |  Should fix https://github.com/PrestaShop/PrestaShop/issues/33524
| Related PRs       | 
| Sponsor company   | 

### How to test

As this is a config specific item, the behavior of the shop should be unchanged. To verify the item, a developer could do:

1. Take `develop` branch and insert the below 2 lines in a Symfony controller to be able to verify the value of the configuration options passed to Doctrine
2. Take this Pull Request branch and do the same, verify that despite the code changes the configuration options passed to Doctrine remain unchanged

2 lines of code
```
$b = $this->get('doctrine.dbal.default_connection');
dump($b->getParams());
```

The behavior is "nothing has changed" because I believe the fix will only fix the situation for a subset of users, victims of https://github.com/PrestaShop/PrestaShop/issues/33524

### CODE REVIEW

**Comment by matks** [comment]
https://github.com/PrestaShop/PrestaShop/commit/4bf12c7af651d68e12fe98e403b26574ec304b2b should fix the CI

**Review by kpodemski — DISMISSED** [review]

**Review by jolelievre — DISMISSED** [review]

**Comment by kpodemski** [comment]
@matks it seems the change is ok but yamllinter isn't happy about the syntax

File: app/config/doctrine.yml [hunk 1]
```diff
@@ -17,9 +17,9 @@ doctrine:
           enum: string
         options:
           # PDO::MYSQL_ATTR_INIT_COMMAND
-          1002: "SET sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''))"
+          !php/const:PDO::MYSQL_ATTR_INIT_COMMAND: "SET sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''))"
```
Comment (kpodemski) [inline app/config/doctrine.yml]:
```suggestion
          !php/const PDO::MYSQL_ATTR_INIT_COMMAND: "SET sql_mode=(SELECT REPLACE(@@sql_mode,'ONLY_FULL_GROUP_BY',''))"
```

File: app/config/doctrine.yml [hunk 2]
```diff
@@ -17,9 +17,9 @@ doctrine:
...
           # PDO::MYSQL_ATTR_MULTI_STATEMENTS
-          1013: '%env(const:runtime:_PS_ALLOW_MULTI_STATEMENTS_QUERIES_)%'
+          !php/const:PDO::MYSQL_ATTR_MULTI_STATEMENTS: '%env(const:runtime:_PS_ALLOW_MULTI_STATEMENTS_QUERIES_)%'
```
Comment (kpodemski) [inline app/config/doctrine.yml]:
```suggestion
          !php/const PDO::MYSQL_ATTR_MULTI_STATEMENTS: '%env(const:runtime:_PS_ALLOW_MULTI_STATEMENTS_QUERIES_)%'
```

**Comment by matks** [comment]
Mmmm maybe I should target 8.1.x ?

**Comment by kpodemski** [comment]
@matks I didn't notice it but it could definitely be part of the next 8 version :D

**Review by kpodemski — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Comment by matks** [comment]
Ah, forgot to rebase this PR on 8.2.x :/ sorry guys. I'll do it in 15 minutes

**Comment by matks** [comment]
Closed in favor of https://github.com/PrestaShop/PrestaShop/pull/36840

**Pull request closed without merging**